Assets/Scripts/Command/DuplicateEntity.cs

```diff
@@ -0,0 +1,46 @@
+﻿using Assets.Scripts.Events;
+using Assets.Scripts.SceneState;
+using System;
+public class DuplicateEntity : Command
+{
+    public override string Name => "Duplicate entity";
+
+    public override string Description => "This command duplicates a selected entity and its children";
+
+    private Guid entityId;
+
+    public DuplicateEntity(Guid entityId)
+    {
+        this.entityId = entityId;
+    }
+
+    public override void Do(DclScene sceneState, EditorEvents editorEvents)
+    {
+        //DclEntity entity = sceneState.SelectionState.PrimarySelectedEntity;
+
+        DclEntity entity = sceneState.GetEntityById(entityId);
+
+        DclEntity newEntity = entity.DeepCopy(sceneState);
```

Comment:
When executing the command, then undoing and then redoing it, this will throw an error. This is because the Guids of the new entities are random, and that causes problems. You should never have any randomness inside a command. If something needs to be randomized, randomize it before the command is called, and store the result of that randomisation in the command.

That means, you need to somehow deterministically generate the Guids inside `DeepCopy`. One way would be by generating the Guids seeded randomly.

See the second answer here: https://stackoverflow.com/questions/13188280/how-to-seed-guid-generation

The seed needs to be actually random generated and stored in the command

Reply:
I have use the method from the second answer to create seeded guids on DclEntity when deep copying, but I have some questions, first one is, which seed should I use every time the method is called and how to save those in the command, so the command know which was the seed used last time. I have commit it here 7e3055ad30886af2f537925be2708bc6d687897d

Reply:
You need to generate a new seed randomly, when a new command is created. You could do this in the Command Factory. Then you need to save the seed in the command in a field. 

Within the `GenerateSeededGuid` method, you shouldn't call `new System.Random(seed)`. This will reset the seeded random generation and always give you the same Guid for a given seed. What you actually want is to parse the `System.Random` to the method. Then you need to call `new System.Random(seed)` in the `Do()` method in the command and parse the random object into `DeepCopy()` as an optional parameter. The method should then look something like this: 
```cs
public DclEntity DeepCopy(DclScene sceneState, System.Random random = null){
    random ??= new System.Random();
    ...
    deepcopyEntity.Id = GenerateSeededGuid(random);
    ...
}
```